daft/expressions/expressions.py

```diff
@@ -5279,122 +5279,38 @@ def mode(self) -> Expression:
         """
         return self.attribute("mode")
 
-    def average_hash(self) -> Expression:
-        """Computes the average hash of an image for deduplication and similarity detection.
+    def hash(self, algorithm: str = "average") -> Expression:
```

Comment:
use a literal type instead of a str. it should provide better typing. 
```py
    def hash(self, algorithm: Literal["average", "perceptual", ...] = "average") -> Expression:
```

Reply:
one other small note. which is no fault of yours, but rather bad timing.  

It appears that you started this PR before this one https://github.com/Eventual-Inc/Daft/pull/5086. Which is deprecating the namespaces in favor of moving things into the `daft.functions` module.

so instead of `col("image").image.hash()` it becomes `image_hash(col("image"))`.

It should a rather minor change to move this in to the `daft.functions` module.